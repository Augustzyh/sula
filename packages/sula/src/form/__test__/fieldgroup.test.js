import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Form, FieldGroup, Field } from '..';
import '../../__tests__/common';


describe('fieldgroup', () => {
  it('dependency', () => {
    const form = (
      <Form
        fields={[
          {
            name: 'group1',
            fields: [
              {
                name: 'input1',
                label: 'input1',
                field: 'input',
              },
              {
                name: 'input2',
                label: 'input2',
                field: 'input',
              },
            ],
          },
          {
            name: 'group2',
            dependency: {
              visible: {
                relates: ['input1'],
                inputs: [['hidden']],
                output: false,
                defaultOutput: true,
              },
            },
            fields: [
              {
                name: 'input3',
                label: 'input3',
                field: 'input',
              },
              {
                name: 'input4',
                label: 'input4',
                field: 'input',
              },
            ],
          },
          {
            dependency: {
              visible: {
                relates: ['input1'],
                inputs: [['hidden']],
                output: false,
                defaultOutput: true,
              },
            },
            fields: [
              {
                name: 'input5',
                label: 'input5',
                field: 'input',
              },
              {
                name: 'input6',
                label: 'input6',
                field: 'input',
              },
            ],
          },
        ]}
      />
    );

    const wrapper = mount(form);
    wrapper
      .find('input')
      .first()
      .simulate('change', { target: { value: 'hidden' } });

    const getFieldGroupInstance = (idx = 0) => wrapper.find(FieldGroup).at(idx).instance();

    expect(getFieldGroupInstance(1).visible).toEqual(true);
    expect(getFieldGroupInstance(2).visible).toEqual(false);
    expect(getFieldGroupInstance(3).visible).toEqual(false);
  });

  it('actions', () => {
    const form = (
      <Form
        fields={[
          {
            name: 'group1',
            fields: [
              {
                name: 'input1',
                label: 'input1',
                field: 'input',
              },
              {
                name: 'input2',
                label: 'input2',
                field: 'input',
              },
            ],
          },
          {
            name: 'group2',
            actionsRender: [
              {
                type: 'button',
                props: {
                  children: 'Btn',
                },
              },
            ],
            fields: [
              {
                name: 'input3',
                label: 'input3',
                field: 'input',
              },
              {
                name: 'input4',
                label: 'input4',
                field: 'input',
              },
              {
                name: 'group2-1',
                fields: [
                  {
                    name: 'input5',
                    label: 'input5',
                    field: 'input',
                  },
                ],
              },
            ],
          },
        ]}
      />
    );

    const wrapper = renderer.create(form).toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('component fieldgroup', () => {
    const form = (
      <Form itemLayout={{ span: 12 }}>
        <FieldGroup>
          <Field name="input" label="input" field="input" initialVisible={false} />
        </FieldGroup>
        <FieldGroup>
          <Field name="input2" label="input2" field="input" />
          <FieldGroup>
            <Field name="input2-1" label="input2" field="input" />
          </FieldGroup>
          <FieldGroup itemLayout={{ span: 6 }}>
            <Field name="input2-2" label="input2" field="input" />
          </FieldGroup>
          <FieldGroup
            actionsRender={{ type: 'button', props: { children: 'btn' } }}
            actionsPosition="default"
          />
          <FieldGroup
            actionsRender={{ type: 'button', props: { children: 'btn' } }}
            actionsPosition="center"
          />
        </FieldGroup>
        <div>
          <Field name="input3" label="input2" field="input" />
        </div>
      </Form>
    );
    expect(renderer.create(form)).toMatchSnapshot();
  });
});
