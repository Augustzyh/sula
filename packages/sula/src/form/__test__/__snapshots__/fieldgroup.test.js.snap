// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fieldgroup actions 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div
    className="ant-spin-container"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input1"
            title="input1"
          >
            input1
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input1"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input2"
            title="input2"
          >
            input2
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input2"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input3"
            title="input3"
          >
            input3
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input3"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input4"
            title="input4"
          >
            input4
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input4"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-8 ant-form-item-label"
          style={Object {}}
        >
          <label
            className=""
            htmlFor="input5"
            title="input5"
          >
            input5
          </label>
        </div>
        <div
          className="ant-col ant-col-12 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <input
                className="ant-input"
                id="input5"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row ant-form-item"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="ant-col ant-col-12 ant-col-offset-8 ant-form-item-control"
          style={Object {}}
        >
          <div
            className="ant-form-item-control-input"
          >
            <div
              className="ant-form-item-control-input-content"
            >
              <div
                className="sula-form-action-default"
              >
                <button
                  className="ant-btn"
                  onClick={[Function]}
                  type="button"
                >
                  <span>
                    Btn
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`fieldgroup component fieldgroup 1`] = `
<div
  className="ant-spin-nested-loading"
>
  <div
    className="ant-spin-container"
  >
    <form
      className="ant-form ant-form-horizontal"
      onSubmit={[Function]}
    >
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "display": "none",
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="input"
                title="input"
              >
                input
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    className="ant-input"
                    id="input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "display": "",
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="input2"
                title="input2"
              >
                input2
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    className="ant-input"
                    id="input2"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "display": "",
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="input2-1"
                title="input2"
              >
                input2
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    className="ant-input"
                    id="input2-1"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-6"
          style={
            Object {
              "display": "",
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="input2-2"
                title="input2"
              >
                input2
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    className="ant-input"
                    id="input2-2"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      >
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "display": "",
              "paddingLeft": 12,
              "paddingRight": 12,
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={
              Object {
                "marginBottom": 0,
              }
            }
          >
            <div
              className="ant-col ant-col-16 ant-col-offset-8 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <div
                    className="sula-form-action-default"
                  >
                    <button
                      className="ant-btn"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        btn
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      />
      <div
        className="sula-form-action-center"
      >
        <button
          className="ant-btn"
          onClick={[Function]}
          type="button"
        >
          <span>
            btn
          </span>
        </button>
      </div>
      <div
        className="ant-row"
        style={
          Object {
            "marginLeft": -12,
            "marginRight": -12,
          }
        }
      />
      <div>
        <div
          className="ant-col ant-col-12"
          style={
            Object {
              "display": "",
            }
          }
        >
          <div
            className="ant-row ant-form-item"
            style={Object {}}
          >
            <div
              className="ant-col ant-col-8 ant-form-item-label"
              style={Object {}}
            >
              <label
                className=""
                htmlFor="input3"
                title="input2"
              >
                input2
              </label>
            </div>
            <div
              className="ant-col ant-col-16 ant-form-item-control"
              style={Object {}}
            >
              <div
                className="ant-form-item-control-input"
              >
                <div
                  className="ant-form-item-control-input-content"
                >
                  <input
                    className="ant-input"
                    id="input3"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
